crates/astria-grpc-mock-test/tests/health/test_response.rs

```diff
@@ -0,0 +1,137 @@
+};
+use futures::future::join;
+use tokio::time::timeout;
+
+use crate::test_utils::start_mock_server;
+
+#[tokio::test]
+async fn default_response_works() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(response::default_response::<HealthCheckResponse>());
+    server.mocked.register(mock).await;
+    let rsp = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "helloworld".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&HealthCheckResponse::default(), rsp.get_ref());
+}
+
+#[tokio::test]
+async fn constant_response_works() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let expected_response = HealthCheckResponse {
+        status: 1,
+    };
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(response::constant_response(expected_response.clone()));
+    server.mocked.register(mock).await;
+    let rsp = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "helloworld".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp.get_ref());
+}
+
+#[tokio::test]
+async fn dynamic_response_works() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let mut expected_response = HealthCheckResponse {
+        status: 1,
+    };
+
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(response::dynamic_response(dynamic_responder));
+    server.mocked.register(mock).await;
+    let rsp_1 = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "1".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp_1.get_ref());
+
+    expected_response.status = 2;
+    let rsp_2 = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "2".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp_2.get_ref());
+}
+
+fn dynamic_responder(request: &HealthCheckRequest) -> HealthCheckResponse {
+    let status_return = request
+        .service
+        .chars()
+        .next()
+        .unwrap()
+        .to_digit(10)
+        .unwrap();
+    HealthCheckResponse {
+        status: i32::try_from(status_return).unwrap(),
+    }
+}
+
+#[tokio::test]
+async fn response_delay_works_as_expected() {
```

Comment:
The test on this file should not be on the mock (as it's now), but on the client receiving its request with a delay (I think this is also how it's implemented in the mock).

I think this delay can be tested simple:

1. mount a mock with any number of responses and a delay of 250ms.
2. run 2 requests against the mock, first with a timeout 200ms, second with 300ms timeout. Observe that the 200ms failed, but the 300ms passes.

Reply:
Done in 4b15aa0.